### PR TITLE
Upgrade to ZIO 2.0

### LINKED ISSUE #463: Upgrade to ZIO 2
ZIO is at Milestone 4, with an RC expected in the next few weeks.
https://github.com/zio/zio/releases/tag/v2.0.0-M4

The API is nearly stable at this point, so any early migration work against this version should pay off towards the official 2.0 release.

The progress is being tracked here:
https://github.com/zio/zio/issues/5470

The Stream Encoding work in progress is the only area where the API might still change before the RC.

To assist with the migration, follow this guide:
https://zio.dev/howto/migrate/zio-2.x-migration-guide/

We are actively working on a ScalaFix rule that will cover the bulk of the simple API changes:
https://github.com/zio/zio/blob/series/2.x/scalafix/rules/src/main/scala/fix/Zio2Upgrade.scala

### PR DESCRIPTION
**Pull request by adamgfraser**
Resolves #463.

### CODE REVIEW

**Comment by adamgfraser** [comment]
This is blocked on `Interop-cats` but otherwise good to go.

**Comment by adamgfraser** [comment]
This is blocked on `Interop-cats` but otherwise good to go.

**Comment by fsvehla** [comment]
As discussed privately: Once interop-cats is ready, push to it’s own topic branch

**Comment by adamgfraser** [comment]
@fsvehla Retargeted to `zio2` branch. This is ready for review.

File: README.md [hunk 1]
```diff
@@ -451,7 +451,7 @@ This attack is very effective in schemas with lots of numbers, causing ops/sec t
 
 # Even Moar Performance
 
-If `zio-json` isn't fast enough for you, then try out [jsoniter-scala](https://github.com/plokhotnyuk/jsoniter-scala); whereas `zio-json` is fully integrated into ZIO, including streams and transducer support, jsoniter is library agnostic.
+If `zio-json` isn't fast enough for you, then try out [jsoniter-scala](https://github.com/plokhotnyuk/jsoniter-scala); whereas `zio-json` is fully integrated into ZIO, including streams and pipeline support, jsoniter is library agnostic.
```
Comment (fsvehla) [inline README.md:454]: Very observant, @adamgfraser

**Review by fsvehla — APPROVED** [review]

**Pull request merged**